nodejs/eks/nodegroup.ts

```diff
@@ -626,3 +626,237 @@ function tagsToAsgTags(tagsInput: InputTags): pulumi.Output<string> {
         return output;
     });
 }
+
+/**
+ * ManagedNodeGroupBaseOptions represents the common configuration settings for
+ * AWS managed NodeGroups.
+ */
+export interface ManagedNodeGroupBaseOptions {
```

Comment:
Is there a reason we need to define a new interface here vs using either our existing NodeGroup interface or the raw AWS input interface?  Is it that we are making some required properties optional?  Can we somehow ensure that if new properties get added to the underlying interface we automatically support them?

Reply:
Great points. I'll try to refactor this to reuse `aws.eks.NodeGroup` instead to support the provider longterm.